packages/mui-material/src/FilledInput/FilledInput.js

```diff
@@ -91,9 +91,16 @@ const FilledInputRoot = styled(InputBaseRoot, {
         // See https://github.com/mui/material-ui/issues/31766
         transform: 'scaleX(1) translateX(0)',
       },
-      [`&.${filledInputClasses.error}:after`]: {
-        borderBottomColor: (theme.vars || theme).palette.error.main,
-        transform: 'scaleX(1)', // error is always underlined in red
+      [`&.${filledInputClasses.error}`]: {
+        [`&:before, &:hover:not(.${filledInputClasses.disabled}):before`]: {
+          borderBottomColor: (theme.vars || theme).palette.error.main,
+        },
+        '&:after': {
+          borderBottomColor: (theme.vars || theme).palette.error.main,
+        },
```

Comment:
What's this part for? looks unrelated to me.

Reply:
Ah, I think I got it.

Reply:
This is to change the color on the bottom border, before focus. Without it it would flicker between blue & red on transform
```
'&:focus-within:after': {
  borderBottomColor: (theme.vars || theme).palette.error.main,
  transform: 'scaleX(1)', // error is always underlined in red
},
```